src/Akka.Streams.Kafka/Messages/CommittableMessage.cs

```diff
@@ -30,21 +41,33 @@ public CommittableMessage(ConsumeResult<K, V> record, CommitableOffset commitabl
     /// should be the next message your application will consume,
     /// i.e. lastProcessedMessageOffset + 1.
     /// </summary>
-    public class CommitableOffset
+    public interface ICommittableOffset : ICommittable
     {
-        private readonly Func<List<TopicPartitionOffset>> _task;
+        PartitionOffset Offset { get; }
+    }
 
-        public CommitableOffset(Func<List<TopicPartitionOffset>> task, PartitionOffset offset)
+    public interface ICommittableOffsetMetadata : ICommittableOffset
+    {
+        string Metadata { get; }
+    }
+
+    
+    public class CommitableOffset : ICommittableOffsetMetadata
+    {
+        private readonly IInternalCommitter _committer;
+        public PartitionOffset Offset { get; }
+        public string Metadata { get; }
+
+        public CommitableOffset(IInternalCommitter committer, PartitionOffset offset, string metadata)
         {
-            _task = task;
+            _committer = committer;
             Offset = offset;
+            Metadata = metadata;
         }
 
-        public PartitionOffset Offset { get; }
-
-        public List<TopicPartitionOffset> Commit()
+        public Task Commit()
         {
-            return _task();
+            return Task.FromResult(_committer.Commit());
```

Comment:
Alpakka's Commit method returns `Future[Done]`, and committed partition offsets can be accessed from `CommittableMessage` directly, so no need to return them from Commit method. Also, in different stages there are different implementations of `IInternalCommitter` `commit `method, so more general `Task` response is better here.

Also, the `Func<List<TopicPartitionOffset>> task` field is replaced with `IInternalCommitter committer` field to allow using additional info for commitment as alpakka does (and we will use this flexibility in `KafkaInternalCommitter` class).